dirs/dirs.go

```diff
@@ -462,11 +469,74 @@ func SetRootDir(rootdir string) {
 		"manjaro-arm",
 	}
 
+	defaultDir := filepath.Join(rootdir, defaultSnapMountDir)
+	altDir := filepath.Join(rootdir, altSnapMountDir)
+
+	// a well known default value
+	dir := "mount-dir-is-unset"
+
+	switch {
```

Comment:
I have a small problem with this. First of all it's different from spec which says we look at the directories and not at the distribution name. I think we should still be entirely directory-driven and re-check against a set of distribution names at the end. If those disagree with the directory-based data we log a big fat warning and carry on with the distribution-name based hard-coded value so that we don't break anyone by accident.

The only exception to this is the super weird initrd with not-quite ID=ubuntu name which I agree we can keep an exception for. If we do that we should also update the spec to say as much (we keep looking at distro and warn if they disagree and we have the special exception for initrd).

Reply:
Thank you for going through this!